File: SingletonExample.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package singletonexample;

/**
 *
 * @author Aaron
 */
public class SingletonExample {
    public static void main(String[] args) {
        PenguinSingleton penguin = PenguinSingleton.INSTANCE;
        System.out.println(penguin.name);
        
        PenguinSingleton anotherPenguin = PenguinSingleton.INSTANCE;
    
        if(anotherPenguin == penguin) {
            System.out.println("They are the same object");
        }
        
        //Note this is not allowed since the constructor is private:
        //PenguinSingleton newPenguin = new PenguinSingleton();
    }
}

File: SingletonExample.cs
using System;

namespace SingletonExample
{
    public sealed class PenguinSingleton
    {
        //The variable will be set to read only
        private static readonly PenguinSingleton instance = new PenguinSingleton();

        private PenguinSingleton() { }

        public string name = "Larry";
        //There will be no instance until a call is made to Instance
        public static PenguinSingleton Instance
        {
            get
            {
                return instance;
            }
        }
    }

    class Program
    {
        static void Main(string[] args)
        {
            //Ask for the penguin instance
            PenguinSingleton Penguin = PenguinSingleton.Instance;
            Console.WriteLine("The penguin's name is " + Penguin.name);

            //Try to make another
            PenguinSingleton AnotherPenguin = PenguinSingleton.Instance;

            //They are the same.
            if(System.Object.ReferenceEquals(Penguin,AnotherPenguin))
            {
                Console.WriteLine("The penguins are the same object");
            }
        }
    }
}

